.ci/mellanox/Dockerfile.centos7

```diff
@@ -0,0 +1,55 @@
+ARG OMPI_CI_OS_NAME=centos
```

Comment:
I think we should put it aside from OMPI repo. This file will be changed after each HPCX release to update MOFED ver and HPCX package name. Also see my comment about building this image only once and pull every PR (below).

IMPORTANT: this changes from testing with not-yet-released HPCX to HPCX GA. IMO it is OK, just highlighting it.

Reply:
After discussion with @amaslenn we decided to exclude docker image building from the Azure pipeline (prepare the image separately and use a generic name for it, e.g. `ompi_ci:latest`.